utils/frame/remote-externalities/src/lib.rs

```diff
@@ -550,154 +523,91 @@ where
 		at: B::Hash,
 		pending_ext: &mut TestExternalities,
 	) -> Result<Vec<KeyValue>, &'static str> {
-		let keys = self.rpc_get_keys_paged(prefix.clone(), at).await?;
+		let start = Instant::now();
+		let mut sp = Spinner::with_timer(Spinners::Dots, "Scraping keys...".into());
+		let keys = self
+			.rpc_get_keys_paged(prefix.clone(), at)
+			.await?
+			.into_iter()
+			.filter(|k| !is_default_child_storage_key(&k.0))
+			.collect::<Vec<_>>();
+		sp.stop_with_message(format!(
+			"✅ Found {} keys ({:.2}s)",
+			keys.len(),
+			start.elapsed().as_secs_f32()
+		));
 		if keys.is_empty() {
 			return Ok(Default::default())
 		}
 
-		let client = self.as_online().rpc_client_cloned();
-		let threads = Self::threads().get();
-		let thread_chunk_size = (keys.len() + threads - 1) / threads;
+		let client = self.as_online().rpc_client();
+		let payloads = keys
+			.iter()
+			.map(|key| ("state_getStorage".to_string(), rpc_params!(key, at)))
+			.collect::<Vec<_>>();
 
-		log::info!(
-			target: LOG_TARGET,
-			"Querying a total of {} keys from prefix {:?}, splitting among {} threads, {} keys per thread",
-			keys.len(),
-			HexDisplay::from(&prefix),
-			threads,
-			thread_chunk_size,
+		let bar = ProgressBar::new(payloads.len() as u64);
+		let bar_message = "Downloading key values".to_string();
+		bar.set_message(bar_message);
+		bar.set_style(
+			ProgressStyle::with_template(
+				"[{elapsed_precise}] {msg} [{wide_bar}] {pos}/{len} ({eta})",
+			)
+			.unwrap()
+			.progress_chars("=>-"),
 		);
+		let payloads_chunked = payloads.chunks(&payloads.len() / Self::PARALLEL_REQUESTS);
+		let requests = payloads_chunked.map(|payload_chunk| {
+			Self::get_storage_data_dynamic_batch_size(
```

Comment:
Yeah it's still impactful. 

Parallel reqs help keep the node working at its maximum capacity returning values, otherwise its cpu usage oscillates (very noticeable when I watch my cpu usage indicator) and overall is considerably slower downloading the key/values. 

Quick run on my laptop running a local node downloading Polkadot key/values:
- Parallel reqs set to 4: 1min 50s
- Parallel reqs set to 1: 2min 43s

I think the performance drop would amplify if using a remote node where higher latency means more time between requests where the node would otherwise be working.

Reply:
Okay thanks for checking.  
Normally the first run is always slower because of missing disk cache, so the impact should be even larger.